### PR TITLE
PKF-642: Add timer start/stop function and enable RTC2 as an extra ti…

### PR DESCRIPTION
**Pull request by minchingho-pax**
…mer resource

Add new timer APIs to turn on/off individual timer for low power application

Test:
This can not be tested by itself. Another review of pax-common-code will be submitted.

### CODE REVIEW

File: hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c [hunk 1]
```diff
@@ -933,3 +958,72 @@ hal_timer_stop(struct hal_timer *timer)
...
+    }
+    __HAL_ENABLE_INTERRUPTS(ctx);
+
+err:
+    return rc;
+}
+/**
+ * Stop a currently running timer tick
+ *
+ * @param tmr The timer to stop
```
Comment (jorajacobi) [inline hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: NIT: @param timer_num

File: hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c [hunk 2]
```diff
@@ -933,3 +958,72 @@ hal_timer_stop(struct hal_timer *timer)
 
     return 0;
 }
+
+/**
+ * Start a currently running timer to tick
+ *
+ * @param tmr The timer to stop
```
Comment (jorajacobi) [inline hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: Same nit as below.

File: hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c (hunk 1 above)
Comment (minchingho-pax) [reply hw/mcu/nordic/nrf52xxx-compat/src/hal_timer.c]: fixed.

**Comment by minchingho-pax** [comment]
Thanks. Going to merge the code.

**Pull request merged**